core/src/main/java/org/mule/api/registry/Registry.java

```diff
@@ -116,8 +117,10 @@
      * @param metadata an implementation specific argument that can be passed into the method
      * @throws RegistrationException if there is a problem unregistering the object. Typically this will be because
      * the object's lifecycle threw an exception
+     * @deprecated as of 3.7.0. Use {@link #unregisterObject(String)} instead
      */
-    void unregisterObject(String key, Object metadata) throws RegistrationException;
+    @Deprecated
+    Object unregisterObject(String key, Object metadata) throws RegistrationException;
```

Comment:
To me this is a deprecation because anyone using this method in its void version won't have any compilation issues because it now returns a value. However, i do want people to stop passing the "metadata" parameter, which was only being used in tests and which honestly makes no sense at all.